modules/host/update.go

```diff
@@ -58,27 +104,182 @@ func (h *Host) ProcessConsensusChange(cc modules.ConsensusChange) {
 	h.mu.Lock()
 	defer h.mu.Unlock()
 
-	for _, block := range cc.RevertedBlocks {
-		// TODO: Look for transactions relevant to open storage obligations.
+	// Wrap the whole parsing into a single large database tx to keep things
+	// efficient.
+	var actionItems []*storageObligation
+	err := h.db.Update(func(tx *bolt.Tx) error {
+		for _, block := range cc.RevertedBlocks {
+			// Look for transactions relevant to open storage obligations.
+			for i, txn := range block.Transactions {
+				// Check for file contracts.
+				if len(txn.FileContracts) > 0 {
+					for _ = range txn.FileContracts {
+						fcid := txn.FileContractID(uint64(i))
+						so, err := getStorageObligation(tx, fcid)
+						if err != nil {
+							// The storage folder may not exist, or the disk
+							// may be having trouble. Either way, we ignore the
+							// problem. If the disk is having trouble, the user
+							// will have to perform a rescan.
+							continue
+						}
+						so.OriginConfirmed = false
+						err = putStorageObligation(tx, so)
+						if err != nil {
+							continue
```

Comment:
if the database is having problems there are bigger issues. Mostly though I wanted to get this all written, there are a few places where the error handling isn't completely correct. But look at the size of this PR. If I had taken the time to do error handling everywhere we'd have added another week or more.

I will revisit this, but right now I think it's more important to get the api integration tests working again, especially since this code isn't consensus critical or anything.